### PR TITLE
feat(buildmasters): Permission support for build masters (CI's)

### PR DESCRIPTION
**Pull request by jervi**
Depends on https://github.com/spinnaker/igor/pull/374, please see that PR for a description.

### CODE REVIEW

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/MonitorJenkinsJobTask.groovy [hunk 1]
```diff
@@ -61,7 +61,8 @@ class MonitorJenkinsJobTask implements OverridableTimeoutRetryableTask {
 
     def buildNumber = (int) stage.context.buildNumber
     try {
-      Map<String, Object> build = buildService.getBuild(buildNumber, master, job)
+      def user = stage.execution.authentication?.user
```
Comment (ajordens) [inline orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/MonitorJenkinsJobTask.groovy]: This should be available on the `ExecutionContext` thread local (and potentially more likely to handle the scenario of their being a manual judgment w/ propagate auth in the middle of the pipeline)

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/BuildService.groovy [hunk 2]
```diff
@@ -33,27 +37,39 @@ class BuildService {
     return UriUtils.encodeFragment(uri.toString(), "UTF-8")
   }
 
-  String build(String master, String jobName, Map<String, String> queryParams) {
-    return igorService.build(master, encode(jobName), queryParams, "")
+  String build(String master, String jobName, Map<String, String> queryParams, String user = null) {
+    return propagateAuth({ igorService.build(master, encode(jobName), queryParams, "") }, user)
```
Comment (ajordens) [inline orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/BuildService.groovy]: Isn't the auth already propagated to these callers (they're just called from regular orca tasks right?)

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/config/IgorConfiguration.groovy [hunk 3]
```diff
@@ -52,11 +53,12 @@ class IgorConfiguration {
   }
 
   @Bean
-  IgorService igorService(Endpoint igorEndpoint, ObjectMapper mapper) {
+  IgorService igorService(Endpoint igorEndpoint, ObjectMapper mapper, RequestInterceptor spinnakerRequestInterceptor) {
     new RestAdapter.Builder()
       .setEndpoint(igorEndpoint)
       .setClient(retrofitClient)
       .setLogLevel(retrofitLogLevel)
+      .setRequestInterceptor(spinnakerRequestInterceptor)
```
Comment (ajordens) [inline orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/config/IgorConfiguration.groovy]: 👍

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/BuildService.groovy (hunk 2 above)
Comment (jervi) [reply orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/BuildService.groovy]: You're right. I've been able to simplify this a lot now, thanks!

**Pull request merged**